frontend/app/src/components/StatusWidget/StatusWidget.test.tsx

```diff
@@ -315,7 +332,18 @@ describe("Running Icon", () => {
       />
     )
 
-    const icon = screen.getByRole("img")
+    const icon = await screen.findByRole("img")
     expect(icon).toHaveAttribute("src", "icon_running.gif")
   })
+
+  it("delays render of running gif", () => {
```

Comment:
in order to double check that something will happen after a delay, it could make sense to advance the timer after the `expect(icon).not.toBeInTheDocument()` check and ensure that afterwards the running man is shown. Otherwise the test could easily be false positive by never render any running gif. What do you think?

Reply:
Yeah, I agree. 

Yesterday I was playing around with the different fake timer options and I found that if I advanced the timer, I still needed to use `findByRole`. But, I didn't like this because `findByRole` will succeed with or without the timer being advanced since it waits for the element to appear. So, I decided to just leave the tests above testing that the running man icon does appear.

I've just tried again adding it as a second step in this test, and in this case `queryByRole` is working. Based on trial and error it looks like the render must be called before advancing the timer and yesterday I was testing only the appearing case separately from the not appearing and advanced the timer at the top of the test.